json-rpc/src/errors.rs

```diff
@@ -10,14 +14,21 @@ use serde_json::Value;
 pub enum ServerCode {
     DefaultServerError = -32000,
 
-    // expose VM status types as server code
+    // VM errors - see `vm_error.rs` for specs
     VmValidationError = -32001,
     VmVerificationError = -32002,
     VmInvariantViolationError = -32003,
     VmDeserializationError = -32004,
     VmExecutionError = -32005,
     VmUnknownError = -32006,
-    // TODO expose Mempool insertion status as server code
+
+    // Mempool errors - see `MempoolStatusCode` for specs
+    MempoolInvalidSeqNumber = -32007,
+    MempoolIsFull = 32008,
```

Comment:
why isn't this one negative?

Are there any tests to ensure that these values make sense?

Reply:
ah that's a really good catch, thanks!

re: testing: Rust doesn't support iterating through enums, even static ones, so to test the enum values we'd either need to use crates like [this](https://docs.rs/strum/0.18.0/strum/ ), or keep a static list of the enum variants to iterate through.. neither of these options seem ideal :/

Reply:
Well, Rust might not, but there's always a macro for that :P

https://docs.rs/enum-iterator/0.6.0/enum_iterator/derive.IntoEnumIterator.html